File: python/prism_sdk/local_protocol_execution_integrity_workflow_fabric.py
"""Choreography P32 local workflow_fabric protocol-compilation integrity feature."""
from .protocol_execution_integrity_support import ProtocolExecutionCard7,ProtocolExecutionRequest4,ProtocolExecutionIntegrityError,manifest,execute
FEATURE_ID="AFA-choreography-P32-F13";CONTRACT_VERSION="choreography-local_protocol_execution_integrity_workflow_fabric/1.0"
def local_protocol_execution_integrity_workflow_fabric_manifest():return manifest(feature_id=FEATURE_ID,contract_version=CONTRACT_VERSION,scale="local",mode="workflow_fabric")
def execute_local_protocol_execution_integrity_workflow_fabric(request:ProtocolExecutionRequest4)->ProtocolExecutionCard7:return execute(request,feature_id=FEATURE_ID,contract_version=CONTRACT_VERSION,scale="local",mode="workflow_fabric")
__all__=["FEATURE_ID","CONTRACT_VERSION","local_protocol_execution_integrity_workflow_fabric_manifest","execute_local_protocol_execution_integrity_workflow_fabric"]

File: crates/choreography/src/local_protocol_execution_integrity_workflow_fabric.rs
//! Choreography P32 local workflow_fabric protocol-execution integrity feature.
use super::protocol_execution_integrity_support::{
    execute, manifest, ProtocolExecutionCard7, ProtocolExecutionIntegrityError,
    ProtocolExecutionRequest4,
};
pub const FEATURE_ID: &str = "AFA-choreography-P32-F13";
pub const CONTRACT_VERSION: &str =
    "choreography-local_protocol_execution_integrity_workflow_fabric/1.0";
pub fn local_protocol_execution_integrity_workflow_fabric_manifest() -> serde_json::Value {
    manifest(FEATURE_ID, CONTRACT_VERSION, "local", "workflow_fabric")
}
pub fn execute_local_protocol_execution_integrity_workflow_fabric(
    request: &ProtocolExecutionRequest4,
) -> Result<ProtocolExecutionCard7, ProtocolExecutionIntegrityError> {
    execute(
        request,
        FEATURE_ID,
        CONTRACT_VERSION,
        "local",
        "workflow_fabric",
    )
}
